util/util.go

```diff
@@ -489,11 +489,71 @@ func PosToLocation(pos token.Pos, pass *analysis.Pass) token.Position {
 }
 
 // ExprToString converts AST expression to string
-func ExprToString(e ast.Expr, pass *analysis.Pass) string {
+func ExprToString(e ast.Expr, pass *analysis.Pass, isShortenExpr bool) string {
 	var buf bytes.Buffer
 	err := printer.Fprint(&buf, pass.Fset, e)
 	if err != nil {
 		panic(fmt.Sprintf("Failed to convert AST expression to string: %v\n", err))
 	}
-	return buf.String()
+	s := buf.String()
+
+	if isShortenExpr {
+		// shorten long expression strings (e.g., s.foo(longVarName, anotherLongVarName, someOtherLongVarName) --> s.foo(...))
+		s = shortenExpr(s)
+	}
+
+	return s
+}
+
+// shortenExpr shortens long expressions enclosed in brackets ('()', '{}', '[]') to "...". It also handles nested brackets
+// and multi-line expressions.
+// Examples:
+// 1. "s.foo(longVarName, anotherLongVarName, someOtherLongVarName)" becomes "s.foo(...)"
+// 2. "s.foo(a, b, c ,d ,e, f).bar(i)" becomes "s.foo(...).bar(i)"
+// 3. "s.foo(someVar, bar(), baz(x, y))" becomes "s.foo(...)"
+// 4. "foo(exprOnLine1,
+// exprOnLine2,
+// exprOnLine3)" becomes "foo(...)"
+func shortenExpr(expr string) string {
+	var result strings.Builder
+	var depth int                 // helps to keep track of nested brackets
+	var inBrackets bool           // indicates if we are currently inside brackets
+	var innerExpr strings.Builder // stores the content inside brackets
+
+	for _, char := range expr {
+		switch char {
+		case '(', '{', '[':
```

Comment:
That's a good point. We could always handle quotes as well in the string processing approach. But I get the broader point that this may require us to deal with many corner cases. As you pointed out, the two options discussed above also have their own issues. So, as we discussed internally, I have updated the shortening logic to do a light-weight AST traversal and printing, with defaulting to `printer` package for all unhandled cases. I have tested it internally and seems to be working fairly well.

Reply:
Also, added many test cases around `(m["("])` :)